narwhal/worker/src/batch_fetcher.rs

```diff
@@ -0,0 +1,530 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+    time::Duration,
+};
+
+use anemo::Network;
+use anyhow::bail;
+use async_trait::async_trait;
+use crypto::NetworkPublicKey;
+use fastcrypto::hash::Hash;
+use futures::{stream::FuturesUnordered, FutureExt, StreamExt};
+use itertools::Itertools;
+use network::WorkerRpc;
+use prometheus::IntGauge;
+use rand::{rngs::ThreadRng, seq::SliceRandom};
+use store::{rocks::DBMap, Map};
+use tracing::debug;
+use types::{Batch, BatchDigest, RequestBatchesRequest, RequestBatchesResponse};
+
+use crate::metrics::WorkerMetrics;
+
+const REMOTE_PARALLEL_FETCH_INTERVAL: Duration = Duration::from_secs(2);
```

Comment:
Using 2s instead of 0.5s as retry interval, to reduce fetch amplifications.

Reply:
This seems a little long? I think the response limit is 6mb currently. So if we assume worst case 10mbps network we should be able to fetch a full payload request @ around .5s. So we should start a parallel request soon after that. Also protects us if the node we are trying to reach is down we would have to wait 2s before the next worker request starts. Maybe we should do 1s?

Reply:
Response limit is 6MB ~ 48mb, but I agree most of the time 0.5s is enough to download it. Right now I'm more worry about destabilizing the host from downloading too much data, which we have observed, vs shorter tail latencies.